internal/controller/promotion/argocd.go

```diff
@@ -217,24 +215,46 @@ func (a *argoCDMechanism) doSingleUpdate(
 	); err != nil {
 		return fmt.Errorf("error patching Argo CD Application %q: %w", app.Name, err)
 	}
-	logging.LoggerFromContext(ctx).WithField("app", app.Name).
-		Debug("patched Argo CD Application")
+	logging.LoggerFromContext(ctx).WithField("app", app.Name).Debug("patched Argo CD Application")
 
-	a.logAppEventFn(ctx, app, "Promotion", "Promotion triggered a sync of this Application resource.")
+	// NB: This attempts to mimic the behavior of the Argo CD API server,
+	// which logs an event when a sync is initiated. However, we do not
+	// have access to the same enriched event data the Argo CD API server
+	// has, so we are limited to logging an event with the best
+	// information we have at hand.
+	//
+	// TODO(hidde): It is not clear what we should do if we have a list of
+	// sources.
+	message := "initiated sync"
+	if app.Spec.Source != nil {
+		message += " to " + app.Spec.Source.TargetRevision
+	}
+	a.logAppEventFn(ctx, app, "kargo-controller", argocd.EventReasonOperationStarted, message)
 
 	return nil
 }
 
-func (a *argoCDMechanism) logAppEvent(ctx context.Context, app *argocd.Application, reason, message string) {
-	logger := logging.LoggerFromContext(ctx)
+func (a *argoCDMechanism) logAppEvent(ctx context.Context, app *argocd.Application, user, reason, message string) {
+	logger := logging.LoggerFromContext(ctx).WithField("app", app.Name)
+
+	// xref: https://github.com/argoproj/argo-cd/blob/44894e9e438bca5adccf58d2f904adc63365805c/server/application/application.go#L2145-L2147
+	// nolint:lll
```

Comment:
The fact that I had to put this here has refueled my dislike of certain linters.